CharlieBackend.Business/Services/ScheduleService.cs

```diff
@@ -0,0 +1,169 @@
+﻿using CharlieBackend.Business.Services.Interfaces;
+using CharlieBackend.Core;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.DTO.Schedule;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using CharlieBackend.Core.Models.ResultModel;
+
+namespace CharlieBackend.Business.Services
+{
+    public class ScheduleService : IScheduleService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public ScheduleService(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<Result<ScheduleDto>> CreateScheduleAsync(CreateScheduleDto scheduleModel)
+        {
+            try
+            {
+                if (scheduleModel == null)
+                {
+                    return Result<ScheduleDto>.Error(ErrorCode.ValidationError, "ScheduleDto is null");
+                }
+
+                var scheduleEntity = _mapper.Map<Schedule>(scheduleModel);
+
+                if (IsNotValidRepeating(scheduleEntity))
+                {
+                    return Result<ScheduleDto>.Error(ErrorCode.ValidationError, "DayNumber can`t be null");
```

Comment:
"DayNumber can't be null"
It doesn't say why it can't be null

Reply:
Added correct error messages in Validate() method